File: Medium/3405-count-the-number-of-special-characters-ii/count-the-number-of-special-characters-ii.py
class Solution:
    def numberOfSpecialChars(self, w: str) -> int:
        n=len(w)

        lower=[-1]*26
        upper=[-1]*26
        for i, ch in enumerate(w):
            if ch.islower():
                lower[ord(ch)-ord('a')]=i
            else:
                if upper[ord(ch)-ord('A')] == -1:
                    upper[ord(ch)-ord('A')]=i

        res=0
        for i in range(26):
            if lower[i] != -1 and lower[i]<upper[i]:
                res+=1

        return res

File: Medium/3405-count-the-number-of-special-characters-ii/count-the-number-of-special-characters-ii.cpp
class Solution{
public:
    int numberOfSpecialChars(string w){
        int n=w.size();

        vector<int> lower(26, -1), upper(26, -1);
        for(int i=0; i<n; i++){
            if(w[i]>='a' && w[i]<='z'){
                lower[w[i]-'a']=i;
            }
            else{
                if(upper[w[i]-'A'] == -1){
                    upper[w[i]-'A']=i;
                }
            }
        }

        int res=0;
        for(int i=0; i<26; i++){
            if(lower[i] != -1 && lower[i]<upper[i]){
                res+=1;
            }
        }

        return res;

    }
};